modules/platforms/cpp/schema/BinaryTupleParser.cpp

```diff
@@ -18,22 +18,61 @@
 #include "BinaryTupleParser.h"
 #include "common/Platform.h"
 
+#include "common/Bytes.h"
+
 #include <cassert>
 #include <cstring>
 #include <stdexcept>
 
+namespace ignite {
+
 namespace {
 
 template <typename T>
-T as(ignite::BytesView bytes) noexcept {
-    T value;
-    std::memcpy(&value, bytes.data(), sizeof(T));
-    return value;
+T loadBytesAs(BytesView bytes, std::size_t offset = 0) noexcept {
+    return bytes::load<Endian::LITTLE, T>(bytes.data() + offset);
 }
 
-} // namespace
+Date loadBytesAsDate(ignite::BytesView bytes) {
+    std::int32_t date = loadBytesAs<std::uint16_t>(bytes);
+    date |= std::int32_t(loadBytesAs<std::int8_t>(bytes, 2)) << 16;
 
-namespace ignite {
+    std::int32_t day = date & 31;
+    std::int32_t month = (date >> 5) & 15;
+    std::int32_t year = (date >> 9); // Sign matters.
+
+    return Date(year, month, day);
+}
+
+Time loadBytesAsTime(ignite::BytesView bytes) {
+    uint64_t time = loadBytesAs<std::uint32_t>(bytes);
+
+    int nano;
+    switch (bytes.size()) {
+        case 4:
+            nano = ((int)time & ((1 << 10) - 1)) * 1000 * 1000;
+            time >>= 10;
+            break;
+        case 5:
+            time |= std::uint64_t(loadBytesAs<std::uint8_t>(bytes, 4)) << 32;
+            nano = ((int)time & ((1 << 20) - 1)) * 1000;
+            time >>= 20;
+            break;
+        case 6:
+            time |= std::uint64_t(loadBytesAs<std::uint16_t>(bytes, 4)) << 32;
+            nano = ((int)time & ((1 << 30) - 1));
+            time >>= 30;
+            break;
+    }
+
+    int second = ((int)time) & 63;
+    int minute = ((int)time >> 6) & 63;
+    int hour = ((int)time >> 12) & 31;
```

Comment:
Using bit shift operation on `int` value implies it have a certain size in bits. And while it's probably true for the most modern systems, why not use sized type here just to be safe?